src/planner/ostree.rs

```diff
@@ -0,0 +1,406 @@
+use crate::{
+    action::{
+        base::{CreateDirectory, CreateFile, RemoveDirectory},
+        common::{ConfigureInitService, ConfigureNix, CreateUsersAndGroups, ProvisionNix},
+        linux::{ProvisionSelinux, StartSystemdUnit, SystemctlDaemonReload},
+        StatefulAction,
+    },
+    error::HasExpectedErrors,
+    planner::{Planner, PlannerError},
+    settings::CommonSettings,
+    settings::{InitSettings, InitSystem, InstallSettingsError},
+    Action, BuiltinPlanner,
+};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+use tokio::process::Command;
+use which::which;
+
+use super::ShellProfileLocations;
+
+/// A planner for Linux installs
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "cli", derive(clap::Parser))]
+pub struct Ostree {
+    /// Where `/nix` will be bind mounted to.
+    #[cfg_attr(feature = "cli", clap(long, default_value = "/var/home/nix"))]
+    persistence: PathBuf,
+    #[cfg_attr(feature = "cli", clap(flatten))]
+    pub settings: CommonSettings,
+    #[cfg_attr(feature = "cli", clap(flatten))]
+    pub init: InitSettings,
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "ostree")]
+impl Planner for Ostree {
+    async fn default() -> Result<Self, PlannerError> {
+        Ok(Self {
+            persistence: PathBuf::from("/var/home/nix"),
+            settings: CommonSettings::default().await?,
+            init: InitSettings::default().await?,
+        })
+    }
+
+    async fn plan(&self) -> Result<Vec<StatefulAction<Box<dyn Action>>>, PlannerError> {
+        let has_selinux = detect_selinux().await?;
+        let mut plan = vec![
+            // Primarily for uninstall
+            SystemctlDaemonReload::plan()
+                .await
+                .map_err(PlannerError::Action)?
+                .boxed(),
+        ];
+
+        plan.push(
+            CreateDirectory::plan(&self.persistence, None, None, 0o0755, true)
+                .await
+                .map_err(PlannerError::Action)?
+                .boxed(),
+        );
+
+        let nix_directory_buf = "\
+                [Unit]\n\
+                Description=Enable mount points in / for ostree\n\
+                ConditionPathExists=!/nix\n\
+                DefaultDependencies=no\n\
+                Requires=local-fs-pre.target\n\
+                After=local-fs-pre.target\n\
+                [Service]\n\
+                Type=oneshot\n\
+                ExecStartPre=chattr -i /\n\
```

Comment:
I'm tempted to fiddle around and see if we can somehow coax systemd-tmpfiles to do this...